python312Packages.app-model: 0.2.8 -> 0.3.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.app-model is: Module to implement generic application schema

meta.homepage for python312Packages.app-model is: https://github.com/pyapp-kit/app-model

meta.changelog for python312Packages.app-model is: https://github.com/pyapp-kit/app-model/blob/v0.3.0/CHANGELOG.md


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0.3.0 with grep in /nix/store/qxnsc72wfsdxp2fac8v7yrlkvjahgg49-python3.12-app-model-0.3.0
- found 0.3.0 in filename of file in /nix/store/qxnsc72wfsdxp2fac8v7yrlkvjahgg49-python3.12-app-model-0.3.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
6 total rebuild path(s)

6 package rebuild(s)

First fifty rebuilds by attrpath

napari
python311Packages.app-model
python311Packages.napari
python312Packages.app-model
python312Packages.napari
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/qxnsc72wfsdxp2fac8v7yrlkvjahgg49-python3.12-app-model-0.3.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.app-model https://github.com/r-ryantm/nixpkgs/archive/ad0706b2981ca163bbc5f5cb8f1141fb612c8c1a.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/ad0706b2981ca163bbc5f5cb8f1141fb612c8c1a#python312Packages.app-model
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/qxnsc72wfsdxp2fac8v7yrlkvjahgg49-python3.12-app-model-0.3.0
ls -la /nix/store/qxnsc72wfsdxp2fac8v7yrlkvjahgg49-python3.12-app-model-0.3.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>napari (python312Packages.napari)</li>
    <li>napari.dist (python312Packages.napari.dist)</li>
    <li>python311Packages.napari</li>
    <li>python311Packages.napari.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.app-model</li>
    <li>python311Packages.app-model.dist</li>
    <li>python312Packages.app-model</li>
    <li>python312Packages.app-model.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @fabaff for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
python312Packages.app-model: 0.2.8 -> 0.3.0

natsukium commented:
Result of `nixpkgs-review pr 342707` at ad0706b2 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/ba091229cad34da7e900fa53f46e155c">napari</a> (python312Packages.napari)</li>
    <li><a href="https://gist.github.com/natsukium/58c7940a80d45372a73818aa6223129c">python311Packages.napari</a></li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.app-model</li>
    <li>python312Packages.app-model</li>
  </ul>
</details>

fabaff committed:
python312Packages.magicgui: 0.5.1 -> 0.9.1

Diff: https://github.com/pyapp-kit/magicgui/compare/refs/tags/v0.5.1...v0.9.1

Changelog: https://github.com/pyapp-kit/magicgui/blob/v0.9.1/CHANGELOG.md

fabaff committed:
python312Packages.napari-npe2: 0.7.2-unstable-2023-10-20 -> 0.7.7

fabaff committed:
python312Packages.napari: 0.4.19.post1 -> 0.5.4

Diff: https://github.com/napari/napari/compare/refs/tags/v0.5.4...v0.5.4

Changelog: https://github.com/napari/napari/releases/tag/v0.5.4

fabaff commented:
Result of `nixpkgs-review pr 342707` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>napari (python312Packages.napari)</li>
    <li>napari.dist (python312Packages.napari.dist)</li>
    <li>python311Packages.app-model</li>
    <li>python311Packages.app-model.dist</li>
    <li>python311Packages.magicgui</li>
    <li>python311Packages.magicgui.dist</li>
    <li>python311Packages.napari</li>
    <li>python311Packages.napari-npe2</li>
    <li>python311Packages.napari-npe2.dist</li>
    <li>python311Packages.napari.dist</li>
    <li>python312Packages.app-model</li>
    <li>python312Packages.app-model.dist</li>
    <li>python312Packages.magicgui</li>
    <li>python312Packages.magicgui.dist</li>
    <li>python312Packages.napari-npe2</li>
    <li>python312Packages.napari-npe2.dist</li>
  </ul>
</details>
